src/shared/fstab-util.c

```diff
@@ -305,19 +305,25 @@ static char *tag_to_udev_node(const char *tagvalue, const char *by) {
 }
 
 char *fstab_node_to_udev_node(const char *p) {
+        const char *q;
```

Comment:
Why not just reuse p?

Reply:
`startswith()` return `NULL` when it does not match. If `NULL` is assigned to `p`, the subsequent checks do not work.